SimPEG/electromagnetics/frequency_domain/survey.py

```diff
@@ -16,13 +16,22 @@ def __init__(self, source_list, **kwargs):
         super(Survey, self).__init__(source_list, **kwargs)
 
         _frequency_dict = {}
+        _source_location_dict = {}
+        _source_location_by_sounding_dict = {}
         for src in self.source_list:
             if src.frequency not in _frequency_dict:
                 _frequency_dict[src.frequency] = []
             _frequency_dict[src.frequency] += [src]
+            if src.i_sounding not in _source_location_dict:
+                _source_location_dict[src.i_sounding] = []
+                _source_location_by_sounding_dict[src.i_sounding] = []
```

Comment:
Would suggest creating your dictionary with the sources directly as keys and location as values, so we can avoid looping again and directly form arrays on `dict.values()`